cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -41,42 +68,97 @@
 @Singleton
 public class StreamSizeScheduler implements Scheduler {
   private static final Logger LOG = LoggerFactory.getLogger(StreamSizeScheduler.class);
+  private static final int STREAM_POLLING_THREAD_POOL_SIZE = 10;
 
+  private final long pollingDelay;
   private final NotificationService notificationService;
   private final StreamAdmin streamAdmin;
   private final StoreFactory storeFactory;
   private final ProgramRuntimeService programRuntimeService;
   private final PreferencesStore preferencesStore;
+  private final ConcurrentMap<Id.Stream, StreamSubscriber> streamSubscribers;
 
+  // Key is scheduleId
+  private final ConcurrentMap<String, StreamSubscriber> scheduleSubscribers;
+
+  private Store store;
+  private Executor notificationExecutor;
+  private ScheduledExecutorService streamPollingExecutor;
 
   @Inject
   public StreamSizeScheduler(CConfiguration cConf, NotificationService notificationService, StreamAdmin streamAdmin,
                              StoreFactory storeFactory, ProgramRuntimeService programRuntimeService,
                              PreferencesStore preferencesStore) {
+    this.pollingDelay = cConf.getLong(Constants.Notification.Stream.STREAM_SIZE_SCHEDULE_POLLING_DELAY);
```

Comment:
Is it sufficient to configure the same polling delay for all streams? Or should it be possible to configure that per stream? Or per schedule?